### PR TITLE
Finished JavaScript & CSS dev-exercises

### PR DESCRIPTION
**Pull request by ldmcdaniel**
Hello:

I have finished the dev-exercises for the front-end JavaScript position.  Please review my code and let me know what you think as I spent quite a bit of time on this (especially the CSS section).

Kind regards,

Lucas McDaniel

### CODE REVIEW

File: css/src/assets/styles/styles.css [hunk 1]
```diff
@@ -0,0 +1,162 @@
+* {
+  padding: 0;
+  margin: 0;
+}
+
+body {
+  background-color: #F5F5F5;
+  background-size: 414px 347px;
+}
+#mainDiv {
```
Comment (dcalhoun) [inline css/src/assets/styles/styles.css:10]: I'd recommend avoiding applying of styles to `id`'s. Using `id`'s for styles can cause specificity headaches down the road when working on a large application.

File: css/src/assets/styles/styles.css [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+div ul:nth-of-type(2) li:nth-of-type(1) {
+  padding-left: 11px;
+  position: absolute;
+  margin-top: 5px;
+  color: #A6A6A6;
+}
+.slideInText {
+  padding: 10px 12px 9px 16px;
+}
+li {
```
Comment (dcalhoun) [inline css/src/assets/styles/styles.css:150]: Styling HTML element directly many times leads to having to override this styles. I'd recommend using a `class` instead.

File: javascript/leap.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+module.exports = isLeapYear;
+
+function isLeapYear (year) {
+  return (
+    (year % 400 === 0) ||
+    (year % 4 === 0 && year % 100 !== 0)
+  ) ? true : false;
```
Comment (dcalhoun) [inline javascript/leap.js:7]: You could potentially shorten this and eliminate the `if/else` conditionals by returning the result of your logical expressions themselves.

File: javascript/phone-number.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+module.exports = PhoneNumber;
+
+function PhoneNumber (num) {
+  num = num.replace(/[(]|[)]|[-]|[\s]|[.]/g, '');
```
Comment (dcalhoun) [inline javascript/phone-number.js:4]: An alternative, more terse approach to consider would be to utilize the `\D` character to replace all non-digits.

File: javascript/phone-number.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  this.number = function () {
+    if (num.length === 11 && num[0] === '1') {
+      num = num.slice(1);
+      return num;
+    }
+    else if (num.length !== 10){
+      num = '0000000000';
+      return num;
+    } else {
```
Comment (dcalhoun) [inline javascript/phone-number.js:13]: :+1: Nice job on the short `if/else` conditional. You could potentially make it shorter by eliminating this last `else` by having your method always return `num`, with your conditional modifying `num` when necessary.

File: javascript/phone-number.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      return num;
+    }
+  };
+  this.areaCode = function () {
+    num = num.slice(0, 3);
+    return num;
+  };
+  this.toString = function () {
+    num = '(' + num.slice(0, 3) + ') '
```
Comment (dcalhoun) [inline javascript/phone-number.js:22]: This could be DRY-er by utilizing your areaCode method to grab the first three digits.

**Comment by dcalhoun** [comment]
Thanks for submitting! Sorry it took a while to provide feedback. Here a couple of additional high-level items to consider:
- When it comes to working on a team, consistency is important. Good job on consistency!
- While JavaScript is sometimes required for animations, I'd recommend avoiding it for simple hover animations when possible.

**Comment by ldmcdaniel** [comment]
David:

Thank you for getting back to me.

Kind regards,

On Wed, Dec 16, 2015 at 11:09 AM, David Calhoun notifications@github.com
wrote:

> Closed #4 https://github.com/HealthTeacher/dev-exercises/pull/4.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/HealthTeacher/dev-exercises/pull/4#event-493722862.

## 

http://ldmcdaniel.com
<+16159264655>       http://linkedin.com/in/ldmcdaniel
http://github.com/ldmcdaniel       http://twitter.com/@ldmcdaniel109

**Pull request closed without merging**